src/SFA.DAS.ApplyService.Web/Startup.cs

```diff
@@ -199,6 +201,8 @@ private void ConfigureDependencyInjection(IServiceCollection services)
             services.AddTransient<ISubmitApplicationConfirmationEmailService, SubmitApplicationConfirmationEmailService>();
             services.AddTransient<ITabularDataService, TabularDataService>();
             services.AddTransient<ITabularDataRepository, TabularDataRepository>();
+            services.AddTransient<IUkprnWhitelistValidator, UkprnWhitelistValidator>();
+            services.AddTransient<IApplyRepository, ApplyRepository>();
```

Comment:
The web layer should not have direct access to SQL, all calls should be made via the API

Reply:
Yes. I thought about it. Shall I open a new end point in  UkrlpLookupController or create a new Controller
das-apply-service\src\SFA.DAS.ApplyService.InternalApi\Controllers\UkrlpLookupController.cs

Reply:
I will use
SFA.DAS.ApplyService.Web\Infrastructure\ApplicationApiClient.cs
SFA.DAS.ApplyService.InternalApi\Controllers\ApplicationController.cs
as they are already consumed by
SFA.DAS.ApplyService.Web\Controllers\Roatp\RoatpApplicationPreambleController.cs

Reply:
I’d put the whitelist lookup into its own api controller, that way it’s following the single responsibility principle and the code can be decommissioned more easily once out of private beta

Reply:
Here is the sequence of processing the whitelisted providers check:
RoatpApplicationPreambleController > UkprnWhitelistValidator > WhitelistedProvidersApiClient > WhitelistedProvidersController > CheckIsUkprnWhitelistedHandler > ApplyRepository